.config/autoconfig.php

```diff
@@ -1,38 +1,26 @@
 <?php
 
+require_once "util.php";
```

Comment:
Not familiar with the autoloading for this or how to test it. Point me somewhere to get started and I'll handle it.

Reply:
As for the first part, are you suggesting that I try to have `getFileEnv` included by renaming the file to x.config.php? Or simply forget the whole idea?

---

As for the database part -

In the entrypoint script those values can independently be from file or not.

https://github.com/nextcloud/docker/blob/405e81528528f7f5a40242ca181ebdfe835e12e7/docker-entrypoint.sh#L162-L164

The improvement here isn't that we allow some to go missing, it is that this PR just so happens to allow the same thing in the PHP. That is,

current behavior (as described in the README section removed by this PR)

- must have postgres_db, _user, _password, _host OR
- postgres_db_file, _user_file, _password_file, _host

updated behavior

- must have one of postgres_db or postgres_db_file AND
- _user or user_file AND
- _password or _password_file AND
- _host

Just noting that any potential confusion from this (e.g. documented by https://github.com/nextcloud/docker/issues/1148#issuecomment-991697378) is additionally avoided by this PR.